reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -426,37 +502,58 @@ public void onSubscribe(Subscription subscription) {
         }
 
         @Override
-        public void onNext(List<ReactiveSocketFactory<SocketAddress>> newFactories) {
-            List<ReactiveSocketFactory<SocketAddress>> removed = computeRemoved(newFactories);
+        public void onNext(Collection<ReactiveSocketFactory<T>> newFactories) {
             synchronized (LoadBalancer.this) {
```

Comment:
Correct.
I'll plan to optimize that in the future (but it didn't show up as a problem on my "limited" stress test)